specs/DOMContentLoaded.md

```diff
@@ -0,0 +1,119 @@
+# Background
+
+In response to consumers' requests for an event similar to the old [WebView DOMContentLoaded](https://docs.microsoft.com/en-us/microsoft-edge/hosting/webview#mswebviewdomcontentloaded), the WebView2 team has introduced DOMContentLoaded API which indicates that the main DOM elements have finished loading.
+In this document we describe the new API. We'd appreciate your feedback.
+
+# Description
+We propose adding DOMContentLoaded to CoreWebView2. This allows the developer to have an event that fires when the DOM is loaded after the WebView2 navigates to a page.
+
+# Examples
+## Win32 C++
+```
+ScenarioDOMContentLoaded::ScenarioDOMContentLoaded(AppWindow* appWindow)
+    : m_appWindow(appWindow), m_webView(appWindow->GetWebView())
+{
+    //! [DOMContentLoaded]
+    // Register a handler for the DOMContentLoaded event.
+    // Event is raised when the DOM content is loaded
+    CHECK_FAILURE(m_webView->add_DOMContentLoaded(
+        Callback<ICoreWebView2DOMContentLoadedEventHandler>(
+            [this](ICoreWebView2* sender, ICoreWebView2DOMContentLoadedEventArgs* args)
+                -> HRESULT {
+                m_webView->ExecuteScript(
```

Comment:
Does the ABI version return an IAsync something that would let the caller build "co_await" chains from C++/WinRT?

Reply:
Yes, although in this case we do not need to await any results.

Reply:
The final parameter for [ExecuteScript](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/0-9-622/icorewebview2#executescript) is an ICoreWebView2ExecuteScriptCompletedHandler* delegate interface. The Invoke method is called on it to provide completion info to the caller.